Source/WebCore/rendering/svg/SVGRenderSupport.cpp

```diff
@@ -55,16 +55,20 @@
 
 namespace WebCore {
 
-LayoutRect SVGRenderSupport::clippedOverflowRectForRepaint(const RenderElement& renderer, const RenderLayerModelObject* repaintContainer)
+LayoutRect SVGRenderSupport::clippedOverflowRectForRepaint(const RenderElement& renderer, const RenderLayerModelObject* repaintContainer, RenderObject::VisibleRectContext context)
 {
     // Return early for any cases where we don't actually paint
     if (renderer.style().visibility() != Visibility::Visible && !renderer.enclosingLayer()->hasVisibleContent())
         return LayoutRect();
 
     // Pass our local paint rect to computeFloatVisibleRectInContainer() which will
     // map to parent coords and recurse up the parent chain.
-    FloatRect repaintRect = renderer.repaintRectInLocalCoordinates();
-    return enclosingLayoutRect(renderer.computeFloatRectForRepaint(repaintRect, repaintContainer));
+    FloatRect repaintRect;
+    if (context.options.contains(RenderObject::VisibleRectContextOption::DoNotUseSVGFastRepaintRectInLocalCoordinates))
+        repaintRect = renderer.repaintRectInLocalCoordinates();
+    else
+        repaintRect = renderer.fastRepaintRectInLocalCoordinates();
```

Comment:
This branching looks inconsistent with other places where we pass the `context`. How difficult is it to merge `fastRepaintRectInLocalCoordinates()` in `repaintRectInLocalCoordinates()` and pass the `context` or another flag to it? 
```
enum class RepaintRectCalculation { Fast, Accurate };
virtual FloatRect repaintRectInLocalCoordinates(RepaintRectCalculation = RepaintRectCalculation::Fast) const;
```
In this case all the changes form calling `repaintRectInLocalCoordinates()` to `fastRepaintRectInLocalCoordinates()` in this patch are not needed.

This call will look like this:
```
auto repaintRect = renderer.repaintRectInLocalCoordinates(context.options.contains(RenderObject::VisibleRectContextOption::CalculateFastRepaintRect) ? RepaintRectCalculation::Fast : RepaintRectCalculation::Accurate);
```

And you can even add a member RepaintRectCalculation to VisibleRectContext instead of it as an option. So this call will look like this:

```
auto repaintRect = renderer.repaintRectInLocalCoordinates(context.calculation);
```

Reply:
That sounds good! Changed.